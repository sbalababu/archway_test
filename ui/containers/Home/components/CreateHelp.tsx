import * as React from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { tomorrowNightEighties } from 'react-syntax-highlighter/dist/styles/hljs';
import Label from './Label';

interface Props {
  host?: string;
  port?: number;
  namespace: string;
}

const SetupHelp = ({ host, port, namespace }: Props) => (
  <div style={{ padding: '4px' }}>
    <Label>step 2. create</Label>
    <p
      style={{
        marginBottom: '0.5em',
        fontSize: '12px',
        textAlign: 'center',
      }}
    >
      create an app that interacts with your workspace
    </p>
    <SyntaxHighlighter language="python" style={tomorrowNightEighties}>
      {`from impala.dbapi
import connect
conn = connect(
  host = '${host || 'worker1.example.com'}',
  port = ${port || '10000'},
  database = '${namespace}'
)
cursor = conn.cursor()
cursor.execute('SELECT * FROM mytable LIMIT 100')
print cursor.description # prints the result setâ€™ s schema
results = cursor.fetchall()`}
    </SyntaxHighlighter>
  </div>
);

export default SetupHelp;
