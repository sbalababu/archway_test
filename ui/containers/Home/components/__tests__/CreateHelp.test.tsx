import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CreateHelp from '../CreateHelp';

describe('CreateHelp', () => {
  it('renders correctly', () => {
    const props = {
      host: 'host',
      port: 8080,
      namespace: 'hdfs',
    };
    const wrapper = shallow(<CreateHelp {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
