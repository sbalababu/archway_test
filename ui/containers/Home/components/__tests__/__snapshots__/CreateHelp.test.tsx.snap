// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateHelp renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "4px",
    }
  }
>
  <Label>
    step 2. create
  </Label>
  <p
    style={
      Object {
        "fontSize": "12px",
        "marginBottom": "0.5em",
        "textAlign": "center",
      }
    }
  >
    create an app that interacts with your workspace
  </p>
  <SyntaxHighlighter
    language="python"
    style={
      Object {
        "hljs": Object {
          "background": "#2d2d2d",
          "color": "#cccccc",
          "display": "block",
          "overflowX": "auto",
          "padding": "0.5em",
        },
        "hljs-addition": Object {
          "color": "#99cc99",
        },
        "hljs-attribute": Object {
          "color": "#ffcc66",
        },
        "hljs-built_in": Object {
          "color": "#f99157",
        },
        "hljs-builtin-name": Object {
          "color": "#f99157",
        },
        "hljs-bullet": Object {
          "color": "#99cc99",
        },
        "hljs-comment": Object {
          "color": "#999999",
        },
        "hljs-deletion": Object {
          "color": "#f2777a",
        },
        "hljs-emphasis": Object {
          "fontStyle": "italic",
        },
        "hljs-keyword": Object {
          "color": "#cc99cc",
        },
        "hljs-link": Object {
          "color": "#f99157",
        },
        "hljs-literal": Object {
          "color": "#f99157",
        },
        "hljs-meta": Object {
          "color": "#f99157",
        },
        "hljs-name": Object {
          "color": "#f2777a",
        },
        "hljs-number": Object {
          "color": "#f99157",
        },
        "hljs-params": Object {
          "color": "#f99157",
        },
        "hljs-quote": Object {
          "color": "#999999",
        },
        "hljs-regexp": Object {
          "color": "#f2777a",
        },
        "hljs-section": Object {
          "color": "#6699cc",
        },
        "hljs-selector-class": Object {
          "color": "#f2777a",
        },
        "hljs-selector-id": Object {
          "color": "#f2777a",
        },
        "hljs-selector-tag": Object {
          "color": "#cc99cc",
        },
        "hljs-string": Object {
          "color": "#99cc99",
        },
        "hljs-strong": Object {
          "fontWeight": "bold",
        },
        "hljs-symbol": Object {
          "color": "#99cc99",
        },
        "hljs-tag": Object {
          "color": "#f2777a",
        },
        "hljs-template-variable": Object {
          "color": "#f2777a",
        },
        "hljs-title": Object {
          "color": "#6699cc",
        },
        "hljs-type": Object {
          "color": "#f99157",
        },
        "hljs-variable": Object {
          "color": "#f2777a",
        },
      }
    }
  >
    from impala.dbapi
import connect
conn = connect(
  host = 'host',
  port = 8080,
  database = 'hdfs'
)
cursor = conn.cursor()
cursor.execute('SELECT * FROM mytable LIMIT 100')
print cursor.description # prints the result setâ€™ s schema
results = cursor.fetchall()
  </SyntaxHighlighter>
</div>
`;
